orangecontrib/text/corpus.py

```diff
@@ -434,8 +434,7 @@ def retain_preprocessing(orig, new, key=...):
                 else:
                     raise TypeError('Indexing by type {} not supported.'.format(type(key)))
                 new._dictionary = orig._dictionary
-
-            new.text_features = orig.text_features
+            new.text_features = list(set(orig.text_features) & set(new.domain.metas)) or None
```

Comment:
Why or `None`? Perhaps it's safer to set `text_features` to an empty list.

Reply:
Perhaps also fix this in `__init__`?

Reply:
`self.text_features = None` changed to `self.text_features = []`
`text_features=None` as a default argument in `__init__` is kept.